lib/outputs/ElasticSearchOutput.js

```diff
@@ -163,98 +131,49 @@ ElasticSearchOutput.prototype._handleOutput = function (eventContainer) {
  * @private
  */
 ElasticSearchOutput.prototype._performPost = function (forced) {
-    var self = this
+    var self = this;
 
     if (self.queue.items.length < self.queueOptions.batchSize && !forced) {
         return
     }
 
     //Copy the queue into here and allow a new queue to fill up
     //TODO: Clean this queueing up!
-    var queue = self.queue
+    var queue = self.queue;
 
     self.queue = {
         items: [],
         write: [],
         timer: null
-    }
+    };
 
-    clearTimeout(queue.timer)
+    clearTimeout(queue.timer);
 
-    var write = ''
+    var bulkBody = [];
 
     for (var eventId in queue.items) {
         //TODO: if event.data[self.timestampField] does not exist we should log a warning
-
-        var event = queue.items[eventId],
-            type = event.data[self.typeField] || self.defaultType,
-            timestamp = event.data[self.timestampField] || new Date(),
-            dateString = timestamp.getFullYear(),
-            month = timestamp.getMonth() + 1,
-            day = timestamp.getDate(),
-            useData = util._extend({}, event.data)
-
-        dateString += '.' + ((String(month).length == 1) ? '0' + month : month)
-        dateString += '.' + ((String(day).length == 1) ? '0' + day : day)
-
-        write += JSON.stringify({ "index": { "_index": self.indexPrefix + '-' + dateString, "_type": type } }) + '\n'
-
-        // Remove fields that elasticseach 2.0 cant have in the source
-        //TODO: include these fields in the index line above?
-        delete useData['_type']
-        delete useData['_uid']
-        delete useData['_id']
-        delete useData['_type']
-        delete useData['_source']
-        delete useData['_all']
-        delete useData['_parent']
-        delete useData['_field_names']
-        delete useData['_routing']
-        delete useData['_index']
-        delete useData['_size']
-        delete useData['_timestamp']
-        delete useData['_ttl']
-
-        write += JSON.stringify(useData) + '\n'
+        var event = queue.items[eventId];
+        bulkBody.push(event.data.bulkActionMeta);
+        // TODO if index / update?
+        bulkBody.push(event.data.message);
     }
 
-    if (write === '') {
-        // Don't bother sending emtpy requests
-        return
+    if(_.size(bulkBody) > 0) {
+        self.elasticClient.bulk(_.extend(self.options, {body: bulkBody})).then(function(response) {
```

Comment:
`util._extend()` would work as well and then you could avoid importing underscore entirely.